From 3587a60d4560329c1133573db6058f942735613b Mon Sep 17 00:00:00 2001
From: Daniela-Marinela Bistrean <daniela.marinela.bistrean@gmail.com>
Date: Fri, 12 Apr 2019 00:49:20 +0300
Subject: [PATCH] Fix integer division issue in int_to_percent function

Changed order of operations to preserve the precision during the division.
---
 src/mixer.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/mixer.c b/src/mixer.c
index d07797a..eb96ae5 100644
--- a/src/mixer.c
+++ b/src/mixer.c
@@ -542,7 +542,7 @@ static int int_to_percent(const struct snd_ctl_elem_info *ei, int value)
     if (range == 0)
         return 0;
 
-    return ((value - ei->value.integer.min) / range) * 100;
+    return ((value - ei->value.integer.min) * 100) / range;
 }
 
 /** Gets a percentage representation of a specified control value.
